app/javascript/services/instrument.service.js

```diff
@@ -0,0 +1,32 @@
+const AudioContext = window.AudioContext || window.webkitAudioContext;
+
+export default class InstrumentService {
+    constructor(sample) {
+        this.sample = sample
```

Comment:
There is a performance improvement we can do: 
At the moment, you are creating a new instance of `AudioContext` for every note played. This consums quite a lot of resources.

Let's instantiate the AudioContext only once in the constructor:

```suggestion
        this.sample = sample
        this.audioCtx = new AudioContext();
```

Then you can reach it using `this.audioCtx` anywhere else in your class ;-)

Reply:
Updated at [81eb0e6](https://github.com/malparty/nimbleat/pull/16/commits/81eb0e68581f202da3f65ca436380cc260e64416)